src/OrleansProviders/LogViews/CustomStorageProvider.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Orleans;
+using Orleans.LogViews;
+using Orleans.Runtime;
+using Orleans.Runtime.LogViews;
+using Orleans.Storage;
+using System.Threading;
+
+namespace Orleans.Providers.LogViews
+{
+    /// <summary>
+    /// A log view provider that relies on grain-specific custom code for 
+    /// loading states from storage, and writing deltas to storage.
+    /// Grains that wish to use this provider must implement the <see cref="ICustomStorageInterface{TState, TDelta}"/>
+    /// interface, to define how state is read and how deltas are written.
+    /// If the provider attribute "PrimaryCluster" is supplied in the provider configuration, then only the specified cluster
+    /// accesses storage, and other clusters may not issue updates. 
+    /// </summary>
+    public class CustomStorageProvider : ILogViewProvider
+    {
+        public string Name { get; private set; }
+
+        public Logger Log { get; private set; }
+
+        private static int counter;
+        private int id;
+
+        public string PrimaryCluster { get; private set; }
+
+        protected virtual string GetLoggerName()
+        {
+            return string.Format("LogViews.{0}.{1}", GetType().Name, id);
+        }
+
+        public Task Init(string name, IProviderRuntime providerRuntime, IProviderConfiguration config)
+        {
+            Name = name;
+            id = Interlocked.Increment(ref counter);
+            PrimaryCluster = config.GetProperty("PrimaryCluster", null);
+
+            Log = providerRuntime.GetLogger(GetLoggerName());
+            Log.Info("Init (Severity={0}) PrimaryCluster={1}", Log.SeverityLevel, 
+                string.IsNullOrEmpty(PrimaryCluster) ? "(none specified)" : PrimaryCluster);
+
+            return TaskDone.Done;
+        }
+
+        public Task Close()
+        {
+            return TaskDone.Done;
+        }
```

Comment:
All providers have this method, it's in the `IProvider` interface - I think it's used to clean up (e.g. close database connections) when the silo shuts down. This provider does not have any cleanup it needs to do, so it just returns.